modules/widgets/instagram.php

```diff
@@ -0,0 +1,497 @@
+<?php
+/**
+ * Plugin Name:  Instagram Widget
+ * Description:  Display some Instagram photos via a widget.
+ * Author:       Automattic Inc.
+ * Author URI:   http://automattic.com/
+ */
+
+use Automattic\Jetpack\Connection\Client;
+
+/**
+ * This is the actual Instagram widget along with other code that only applies to the widget.
+ */
+class Jetpack_Instagram_Widget extends WP_Widget {
+
+	const ID_BASE = 'wpcom_instagram_widget'; // Don't change this as Atomic widgets will break
+
+	public $valid_options;
+	public $defaults;
+
+	/**
+	 * Sets the widget properties in WordPress, hooks a few functions, and sets some widget options.
+	 */
+	function __construct() {
+		parent::__construct(
+			self::ID_BASE,
+			__( 'Instagram', 'jetpack' ),
+			array(
+				'description' => __( 'Display your latest Instagram photos.', 'jetpack' ),
+			)
+		);
+
+		add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_css' ) );
+		add_action( 'wp_ajax_wpcom_instagram_widget_update_widget_token_id', array( $this, 'ajax_update_widget_token_id' ) );
+
+		$this->valid_options = array(
+			'max_columns' => 3,
+			'max_count'   => 20,
+		);
+
+		$this->defaults = array(
+			'token_id' => null,
+			'title'    => __( 'Instagram', 'jetpack' ),
+			'columns'  => 2,
+			'count'    => 6,
+		);
+	}
+
+	/**
+	 * Enqueues the widget's frontend CSS but only if the widget is currently in use.
+	 */
+	public function enqueue_css() {
+		if ( ! is_active_widget( false, false, self::ID_BASE ) )
```

Comment:
Could you add the 2 other use cases we usually handle in other widgets, like here:
https://github.com/Automattic/jetpack/blob/8c01e22d807f16dcb8d3de73fcf3b0e8641344d8/modules/widgets/facebook-likebox.php#L46-L48

`is_active_widget( false, false, 'monster' ) || is_customize_preview()`